packages/models-library/src/models_library/osparc_variable_identifier.py

```diff
@@ -83,8 +97,11 @@ def replace_osparc_variable_identifier(  # noqa: C901
         if obj.default_value is not None:
             return deepcopy(obj.default_value)  # type: ignore
     elif isinstance(obj, dict):
-        for key, value in obj.items():
-            obj[key] = replace_osparc_variable_identifier(value, osparc_variables)
+        for key, value in list(obj.items()):
+            del obj[key]
```

Comment:
I find your observation very funny.
This is the approach I discarded in favour of the current, because it is way more readable and less buggy/error prone.

Reply:
Not sure i get why my observation is funny :-) 
I think your code is much more error prone than my alternative. Specially if you accidentaly remove the `list`.